### PR TITLE
HPCC-20212 Move new Graph into GraphTreeWidget paradigm

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes HPCC-20212

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq Please review

@richardkchapman this PR is not ready for pulling as it is dependent on some code reviews + a release in the hpcc-js libraries.

File: esp/src/eclwatch/dojoConfig.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 var dojoConfig;
+const debugHPCC_JS = true; //  Should never be TRUE in a PR  ---
```
Comment (GordonSmith) [inline esp/src/eclwatch/dojoConfig.js]: Revert to false before final pull.

File: esp/src/package.json [hunk 2]
```diff
@@ -22,26 +22,17 @@
...
-    "@hpcc-js/form": "0.0.81",
-    "@hpcc-js/graph": "0.1.1",
-    "@hpcc-js/layout": "0.1.6",
-    "@hpcc-js/map": "0.2.0",
-    "@hpcc-js/other": "0.2.0",
-    "@hpcc-js/timeline": "0.0.81",
-    "@hpcc-js/tree": "0.0.79",
+    "@hpcc-js/chart": "0.8.10",
+    "@hpcc-js/comms": "0.4.14",
+    "@hpcc-js/eclwatch": "0.2.20",
```
Comment (GordonSmith) [inline esp/src/package.json]: Bump version to latest once hpcc-js is released.

**Comment by richardkchapman** [comment]
@miguelvazq

File: esp/src/src/Utility.ts [hunk 3]
```diff
@@ -23,6 +23,12 @@ export function xmlEncode2(str) {
         ;
 }
 
+export function decodeHtml(html) {
```
Comment (miguelvazq) [inline esp/src/src/Utility.ts]: Wouldn't this be better titled "decodeTextAreaHtml"?

**Review by miguelvazq — COMMENTED** [review]
Minor comment.

File: esp/src/src/Utility.ts (hunk 3 above)
Comment (GordonSmith) [reply esp/src/src/Utility.ts]: @miguelvazq not really its a general purpose HTML -> Text decoder which happens to use a textarea (internally) to do the heavy lifting.

**Pull request merged**